### PR TITLE
[souschef] Introduce unified gaussian generator

### PR DESCRIPTION
**Pull request by binarman**
This commit adds unified gaussian generator of tensor data.

related issue #6576 

ONE-DCO-1.0-Signed-off-by: Alexander Efimov <a.efimov@samsung.com>

### CODE REVIEW

File: compiler/souschef/src/Gaussian.cpp [hunk 1]
```diff
@@ -26,6 +26,43 @@
...
+    {
+      res.emplace_back(arr[b]);
+    }
+  }
+
+  return res;
+}
+
+template <typename T>
+static std::vector<uint8_t> generate_gaussian(int32_t count, float mean, float stddev)
```
Comment (binarman) [inline compiler/souschef/src/Gaussian.cpp:56]: There is TODO related to seed, so I added this method.

Maybe we can make generator methods public and test them. Though not sure how to test random generators properly now. =/

File: compiler/souschef/src/Gaussian.cpp [hunk 2]
```diff
@@ -26,6 +26,44 @@
...
+  std::normal_distribution<float> dist{mean, stddev};
+
+  std::vector<uint8_t> res;
+
+  constexpr float max_cap = std::numeric_limits<T>::max();
+  constexpr float min_cap = std::numeric_limits<T>::min();
+  for (uint32_t n = 0; n < count; ++n)
+  {
+    float raw_value = dist(rand);
+    const float capped_value = std::max(min_cap, std::min(max_cap, raw_value));
```
Comment (binarman) [inline compiler/souschef/src/Gaussian.cpp:43]: This "capping" is just for reinsurance of correct behavior, standard says: "The behavior is undefined if the truncated value cannot be represented in the destination type." about float -> int conversion.

Though it prbably will work anyway.

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Comment by seanshpark** [comment]
Please merge after test passes or request another reviewer: I'm on a edu course today -_-;

**Comment by binarman** [comment]
> Please merge after test passes or request another reviewer: I'm on a edu course today -_-;

Thank you! Have a good day then =)

**Pull request merged**